### ISSUE TITLE
Adding save and load support

### ISSUE THREAD

**Issue by rimvydasrub** [peterdavidfagan/mujoco_robot_environments#8]
Currently, its not straightforward how the save and load functionality works for the environment. This functionality is valuable for:

- replication of the results as just setting random seed (``random.seed(###)``) is not sufficient to fix randomness in the system
- go generate environments and do feature prepossessing offline so that training and inference can be done in a faster manner.

**Comment by peterdavidfagan** [comment]
Can you provide further details, what are you asking for in the API? What are you looking to save/load? 

Reproducibility and random seeds is something I can look into at some point.

**Comment by rimvydasrub** [comment]
If you run the following program

```python
import random

from PIL import Image
import numpy as np
import torch
from mujoco_robot_environments.tasks.rearrangement import RearrangementEnv

# setting random seeds
random.seed(0)

env = RearrangementEnv(viewer=True)
env.reset()

obs = env._compute_observation()['overhead_camera/rgb']
obs_img = Image.fromarray(obs)
obs_img.save('obs.png')

env.close()

```
you get two different results 
![obs](https://github.com/peterdavidfagan/mujoco_robot_environments/assets/15063962/a2057b81-8151-4772-bb1e-5d6f0ce14df3)
![obs-1](https://github.com/peterdavidfagan/mujoco_robot_environments/assets/15063962/382e9f8f-0446-49e0-bffa-920e15223aa8)

After inspection, this is because ``np.random.seed`` is not set explicitly. I can now replicate environments with the same seed.

Generally, for the API functionality, is something like this possible?

```python
from mujoco_robot_environments.tasks.rearrangement import RearrangementEnv

env = RearrangementEnv(viewer=True)
env.reset()

# saving the environment
env.save("save_file_name")

# loading the environment
env = RearrangementEnv.load("save_file_name")

```
